crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -84,6 +84,23 @@ fn remove_children(parent: Entity, children: &[Entity], world: &mut World) {
     }
 }
 
+fn clear_children(parent: Entity, world: &mut World) {
+    if let Some(children) = world.entity(parent).get::<Children>() {
+        remove_children(parent, &children.0.clone(), world);
```

Comment:
While code reuse here is nice, cloning here is suboptimal, as it potentially heap allocates for a temporary.

We could just iterate over the existing children, remove the Parent components from them, then just clear the Children component. This would avoid the allocation entirely, at the cost of some code repetition.

Reply:
Agreed, but rather that clearing afterwards we can use `remove` instead of `get`.

Reply:
Done